docs/learn/encyclopedia/transactions-specialized/channel-accounts.mdx

```diff
@@ -643,94 +636,174 @@ Here's an example of what that might looks like:
+    # Partial failure logic example
+    for op in bundle.operations:
+      if isinstance(op, Asset):  # Assuming only payments
+        failedRecipients.append(op.destination)
+  finally:
+    channelData["state"] = "available"
 ```
 
 ```js
-// Function to find an available channel account
-function getAvailableChannelAccount() {
-  return channelAccounts.find((account) => account.state === "available");
+static class ChannelData {
+  AccountResponse account;
+  KeyPair keypair;
+  String state;
+
+  ChannelData(AccountResponse account, KeyPair keypair, String state) {
+    this.account = account;
+    this.keypair = keypair;
+    this.state = state;
+  }
 }
 
-// Function to mark channel account as in use
-function markChannelAccountInUse(channelAccount) {
-  channelAccount.state = "in use";
+// Dynamic recipients' public keys
+String[] allRecipients = {
+  "GD72...B2D", "GDX7...G7M", "GBI3...Q4V", "GBH2...N6E", "GCJY...OJ2", ...
+};
+
+List<ChannelData> channelAccountsTracker = new ArrayList<>();
+for (int i=0; i < channelAccounts.size(); i++) {
+  channelAccountsTracker.add(
+    new ChannelData(
+      channelAccounts.get(i),
+      channelKeypairs.get(i),
+      "available")
+    );
 }
 
-// Function to mark channel account as available
-function markChannelAccountAvailable(channelAccount) {
-  channelAccount.state = "available";
+List<Transaction> txOutput = new ArrayList<>();
+int recipientIndex = 0;
+
+// Build transactions dynamically based on available recipients
+while (recipientIndex < allRecipients.length) {
+    for (ChannelData channelData : channelAccountsTracker) {
+        if (recipientIndex >= allRecipients.length) {
+            break;
+        }
+        if (channelData.state.equals("available")) {
+            channelData.state = "active";
+            Transaction.Builder txBundleBuilder = new Transaction.Builder(channelData.account, Network.TESTNET)
+                    .setBaseFee(640)
+                    .setTimeout(3600);
+
+            // Add up to 100 payment operations per channel
+            for (int j = 0; j < 100; j++) {
+                if (recipientIndex >= allRecipients.length) {
+                    break;
+                }
+                PaymentOperation paymentOperation = new PaymentOperation.Builder(
+                        allRecipients[recipientIndex],
+                        AssetTypeNative.INSTANCE,
+                        "10"
+                ).setSourceAccount(primaryKeypair.getAccountId()).build();
+                txBundleBuilder.addOperation(paymentOperation);
+                recipientIndex++;
+            }
+
+            Transaction txBundle = txBundleBuilder.build();
+            txBundle.sign(primaryKeypair);
+            txBundle.sign(channelData.keypair);
+
+            txOutput.add(txBundle);
+        } else {
+            System.out.println("No available channel account. Waiting...");
+            // Perhaps sleep here to wait for availability
```

Comment:
Should be console.log for stdout.

Reply:
Looks like I had the wrong language config'd here, so I moved it in 09fffca9a31f5a056c72ede2c8b5486721030f52. Thank you for noticing the inconsistency! 💜 I try to minimize the print statements to crucial execution points since I assume most people just read it for a pseudocode understanding. 💭